### ISSUE TITLE
[BUG] Rule matching on name, not on fully qualified name

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#405]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0-beta2

### Priority

Critical

### Current Behavior

See the following image, in which rule `hibernate-search-00750`, which should match on `org.hibernate.search.annotations.Field` (as an `ANNOTATION`) is being matched on `java.lang.reflect.Field` as a parameter passed to a method.
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/4ab6ed1c-76df-4788-add9-b2500b7a5185)

### Expected Behavior

- The rule should match on the fully qualified name, not on the name of the class.
- The rule should match on the annotation usage of the given class.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Run kantra with the following config:
```
./kantra analyze --target quarkus --target jakarta-ee --target openjdk17 --target cloud-readiness --target eap8 --target azure-appservice --target camel4 --analyze-known-libraries
```
2. In the issues, search for "@Field"
3. See where the issue has been highlighted

### Environment

```markdown
- Fedora 38
- Kantra (0.3.0-beta2)
- camunda-bpm-spring-boot-starter-example-war-1.3.0.war from [40SampleApps](https://drive.google.com/file/d/1yKkFtwGoMJJUBgIinZUCZ3nFmlDs2rYv/view)
```

**Comment by jmle** [comment]
Also, in the same report, look for `javax.xml.soap.SOAPElementFactory.newInstance() must be replaced` so see another instance of this happening

**Comment by shawn-hurley** [comment]
I believe the issue here, is similar, in that we do not use exact matches for ANNOTATIONS.

When I use location: PACKAGE or TYPE, I can see no matches for this rule.

@jmle, can you verify that the expected output is that this does not match?

**Comment by jmle** [comment]
@shawn-hurley with location `TYPE` the rule doesn't match; `org.hibernate.search.annotations.Field` is not being used anywhere in the code.

**Comment by shawn-hurley** [comment]
I'd like to know if we can add the package search w/to the annotation to see if this changes something. 

Now that I have a test case, I can see I will get something up hopefully today.

**Comment by jmle** [comment]
Something has changed and I cannot reproduce this on the latest commit, but I can on `abc863b`. Investigating.

**Comment by shawn-hurley** [comment]
@jmle (just because this is assigned to me), did anything ever turn up here? Based on my testing, I think it should be working as expected, but I wanted to double-check with you.

**Comment by shawn-hurley** [comment]
Closing this issue, as it appears to be working, we can re-open if we need to
